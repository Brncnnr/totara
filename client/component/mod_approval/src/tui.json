{
  "component": "mod_approval",
  "vendor": "totara",
  "dependencies": [
    "tui_xstate",
    "totara_engage",
    "totara_comment",
    "totara_notification",
    "editor_weka"
  ]
}
