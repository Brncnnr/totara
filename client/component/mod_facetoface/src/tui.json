{
  "component": "mod_facetoface",
  "vendor": "totara"
}
