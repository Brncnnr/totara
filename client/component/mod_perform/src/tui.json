{
  "component": "mod_perform",
  "vendor": "totara",
  "dependencies": [
    "totara_notification"
  ]
}
