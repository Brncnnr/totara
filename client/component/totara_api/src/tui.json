{
  "component": "totara_api",
  "vendor": "totara"
}