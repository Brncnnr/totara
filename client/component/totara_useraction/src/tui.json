{
  "component": "totara_useraction",
  "vendor": "totara"
}
