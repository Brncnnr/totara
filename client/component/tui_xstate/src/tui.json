{
  "component": "tui_xstate",
  "vendor": "totara",
  "entry": "./js/internal/entry.js"
}
