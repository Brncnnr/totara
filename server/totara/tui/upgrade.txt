This file describes API changes in /tui/ information provided here is intended
especially for developers.

=== 16.0 ===

* Added a check to bail out early instead of hanging when compiling large non-ASCII SCSS files under PHP < 7.4, and added the pre_php74_allow_unperformant_css plugin setting to allow switching this off and attempting to compile anyway.

=== 15.0 ===

* Changed type of theme parameter from param_alphaext! to param_component! for query totara_tui_themes_with_variables.

=== 14.0 ===

* tenantid renamed to tenant_id in totara_tui\controllers\theme_settings