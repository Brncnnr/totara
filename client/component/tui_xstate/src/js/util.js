/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2021 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Simon Tegg <simon.tegg@totaralearning.com>
 * @module tui_xstate
 */
import { isDraft } from 'tui/immutable';
import { createSelectorCreator, defaultMemoize } from 'reselect';
import { showError as tuiShowError } from 'tui/errors';
export const createSelector = createSelectorCreator(customMemoize);

/**
 * Custom createSelector wrapper.
 *
 * Will bypass the selector memoization/cache when passed a draft object.
 */

/**
 * Custom memoize -- bypass the memoization when passed a draft object to avoid
 * invalidating the cache.
 */
function customMemoize(func, equalityCheckOrOptions) {
  const defaultMemoized = defaultMemoize(func, equalityCheckOrOptions);
  const memoized = function(...args) {
    return args.some(x => isDraft(x))
      ? func.apply(null, arguments)
      : defaultMemoized.apply(null, arguments);
  };
  memoized.clearCache = defaultMemoized.clearCache;
  return memoized;
}

/**
 * XState wrapper for Tui showError Modal
 *
 * @param {object} context
 * @param {import('xstate').Event} event
 */
export function showError(context, event) {
  if (event.data instanceof Error) {
    tuiShowError(event.data);
    return;
  }

  if (event.type == 'xstate.error' && event.data.data instanceof Error) {
    tuiShowError(event.data.data);
    return;
  }

  console.error(
    `[Tui XState] tui_xstate/util showError passed an event without an Error.
    Original params:`,
    { context, event }
  );
}
