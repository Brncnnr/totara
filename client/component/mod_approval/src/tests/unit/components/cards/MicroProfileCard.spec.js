/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2021 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Tatsuhiro Kirihara <tatsuhiro.kirihara@totaralearning.com>
 * @module mod_approval
 */

import { shallowMount } from '@vue/test-utils';
import { structuralDeepClone } from 'tui/util';
import component from 'mod_approval/components/cards/MicroProfileCard.vue';

const props = {
  user: {
    id: 42,
    fullname: 'Totara User',
    email: 'totara.user@example.com',
    profileimageurl: 'https://example.com/user.jpg',
  },
};

describe('mod_approval/components/cards/MicroProfileCard', () => {
  let propsData;

  beforeEach(() => {
    propsData = structuralDeepClone(props);
  });

  it('Check snapshot with default', () => {
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check snapshot with large size', () => {
    propsData.size = 'large';
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check snapshot with readonly', () => {
    propsData.readOnly = true;
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check snapshot with custom name', () => {
    propsData.name = 'Cus Tom Name';
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check snapshot with email', () => {
    propsData.showEmail = true;
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check snapshot with emphasised', () => {
    propsData.emphasiseName = true;
    const wrapper = shallowMount(component, { propsData });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Check avatarUrl', () => {
    let wrapper;

    delete propsData.user.profileimageurlsmall;
    delete propsData.user.profileimageurl;
    delete propsData.user.card_display;

    propsData.user.profileimageurlsmall = 'https://example.com/small.jpg';
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.vm.avatarUrl).toEqual('https://example.com/small.jpg');

    propsData.user.profileimageurl = 'https://example.com/medium.jpg';
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.vm.avatarUrl).toEqual('https://example.com/medium.jpg');

    propsData.user.card_display = {
      profile_picture_url: 'https://example.com/large.jpg',
    };
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.vm.avatarUrl).toEqual('https://example.com/large.jpg');
  });

  it('Check avatarAlt', () => {
    let wrapper;

    wrapper = shallowMount(component, { propsData });
    expect(wrapper.vm.avatarAlt).toEqual('');

    propsData.user.card_display = { profile_picture_alt: 'DJ ToT@rA' };
    wrapper = shallowMount(component, { propsData });
    expect(wrapper.vm.avatarAlt).toEqual('DJ ToT@rA');
  });
});
