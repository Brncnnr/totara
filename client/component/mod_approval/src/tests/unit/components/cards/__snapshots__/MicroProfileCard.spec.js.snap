// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with custom name 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--xsmall"
>
  <avatar-stub
    alt=""
    size="xsmall"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <a
      class="tui-mod_approval-microProfileCard__name tui-mod_approval-microProfileCard__name--link"
      href="http://foo/user/profile.php?id=42"
    >
      
      Cus Tom Name
    
    </a>
     
    <!---->
  </div>
</div>
`;

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with default 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--xsmall"
>
  <avatar-stub
    alt=""
    size="xsmall"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <a
      class="tui-mod_approval-microProfileCard__name tui-mod_approval-microProfileCard__name--link"
      href="http://foo/user/profile.php?id=42"
    >
      
      Totara User
    
    </a>
     
    <!---->
  </div>
</div>
`;

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with email 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--xsmall"
>
  <avatar-stub
    alt=""
    size="xsmall"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <a
      class="tui-mod_approval-microProfileCard__name tui-mod_approval-microProfileCard__name--link tui-mod_approval-microProfileCard__name--primary"
      href="http://foo/user/profile.php?id=42"
    >
      
      Totara User
    
    </a>
     
    <a
      class="tui-mod_approval-microProfileCard__email"
      href="mailto:totara.user@example.com"
    >
      totara.user@example.com
    </a>
  </div>
</div>
`;

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with emphasised 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--xsmall"
>
  <avatar-stub
    alt=""
    size="xsmall"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <a
      class="tui-mod_approval-microProfileCard__name tui-mod_approval-microProfileCard__name--link tui-mod_approval-microProfileCard__name--primary"
      href="http://foo/user/profile.php?id=42"
    >
      
      Totara User
    
    </a>
     
    <!---->
  </div>
</div>
`;

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with large size 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--large"
>
  <avatar-stub
    alt=""
    size="large"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <a
      class="tui-mod_approval-microProfileCard__name tui-mod_approval-microProfileCard__name--link"
      href="http://foo/user/profile.php?id=42"
    >
      
      Totara User
    
    </a>
     
    <!---->
  </div>
</div>
`;

exports[`mod_approval/components/cards/MicroProfileCard Check snapshot with readonly 1`] = `
<div
  class="tui-mod_approval-microProfileCard tui-mod_approval-microProfileCard--xsmall"
>
  <avatar-stub
    alt=""
    size="xsmall"
    src="https://example.com/user.jpg"
  />
   
  <div
    class="tui-mod_approval-microProfileCard__profile"
  >
    <span
      class="tui-mod_approval-microProfileCard__name"
    >
      Totara User
    </span>
     
    <!---->
  </div>
</div>
`;
